validator/src/main.rs

```diff
@@ -3072,3 +3070,29 @@ fn process_account_indexes(matches: &ArgMatches) -> AccountSecondaryIndexes {
         indexes: account_indexes,
     }
 }
+
+/// Get the Slot and Hash of the local snapshot with the highest slot.  Can be either a full
+/// snapshot or an incremental snapshot.
+fn get_highest_local_snapshot_hash(
```

Comment:
Getting info about the highest local snapshot was used in two places, so I refactored 'em into this function. One needs the Slot and Hash, whereas the other only needs the Slot. I don't think the extra Hash will make a performance impact here.